lib/common/api/deprecate.js

```diff
@@ -69,23 +69,28 @@ const deprecate = {
     })
   },
 
-  promisify: (promise, cb) => {
-    const oldName = `function with callbacks`
-    const newName = `function with Promises`
-    const warn = warnOnce(oldName, newName)
+  promisify: (fn, numArgs) => {
+    return function (...args) {
+      const promise = fn.apply(this, args.splice(0, numArgs))
+      const cb = args[numArgs]
 
-    if (typeof cb !== 'function') return promise
-    if (process.enablePromiseAPIs) warn()
-    return promise
-      .then(res => {
-        process.nextTick(() => {
-          cb(null, res)
-        })
-      }, err => {
-        process.nextTick(() => {
-          cb(err)
+      const oldName = `function with callbacks`
+      const newName = `function with Promises`
+      const warn = warnOnce(oldName, newName)
```

Comment:
_(bug)_ This is creating a new `warn ` each time this return function is called -- so wouldn't a warning be given every time our wrapper is function is called, rather than just the first time?

To give the warning just once, I think we'd want a `warn` to be created each time `deprecate.promisify()` is called, *not* each time the wrapper function is called, e.g.:

```js
promisify: (fn, numArgs) => {
  const oldName = ...
  const newName= ...
  const warn = warnOnce(oldName, newName)

  return function (...args) {
    const promise= fn.apply(this, args.splice(0, numArgs))
    ...
  }
}
```